yt_dlp/extractor/bbc.py

```diff
@@ -438,41 +438,76 @@ def _process_media_selector(self, media_selection, programme_id):
                 subtitles = self.extract_subtitles(media, programme_id)
         return formats, subtitles
 
-    def _download_playlist(self, playlist_id):
-        try:
-            playlist = self._download_json(
-                f'http://www.bbc.co.uk/programmes/{playlist_id}/playlist.json',
-                playlist_id, 'Downloading playlist JSON')
+    def _download_playlist(self, playlist_id, webpage):
+        playlist = self._download_json(
+            f'http://www.bbc.co.uk/programmes/{playlist_id}/playlist.json',
+            playlist_id, 'Downloading playlist JSON')
+
+        subtitles = {}
+        programme_id = None
+        title = None
+        description = None
+        duration = None
+
+        def _parse_formats(programme_id, types=None):
             formats = []
-            subtitles = {}
+            version_formats, version_subtitles = self._download_media_selector(programme_id)
+            types = types or 'Original'
+            for f in version_formats:
+                f['format_note'] = types
+                if any('AudioDescribed' in x for x in types):
+                    f['language_preference'] = -10
+
+            formats += version_formats
+            for tag, subformats in (version_subtitles or {}).items():
+                subtitles.setdefault(tag, []).extend(subformats)
+            return formats, subtitles
 
+        # Fallback when there is no VPID in json response.
+        if not traverse_obj(playlist, ('defaultAvailableVersion')):
+            episode_block = self._search_regex(
+                r'"episode"\s*:\s*\{(.*?)\}\s*,\s*"relatedEpisodes"',
+                webpage,
+                'episode',
+                fatal=False,
+            )
+            episode_json = {}
+            if episode_block:
+                episode_json = self._parse_json('{' + episode_block + '}', playlist_id)
+
+            description = traverse_obj(episode_json, ('synopses', 'large'))
+            programme_id = self._search_regex(
+                r'"id"\s*:\s*"([0-9a-z]{8})"',
+                self._search_regex(
+                    r'"versions"\s*:\s*\[\s*(\{[^]]*"id"\s*:\s*"[0-9a-z]{8}".*?)\]\s*,',
```

Comment:
again, I don't like regex searching inside JSON like this. it's brittle and error-prone